src/objectManager.cpp

```diff
@@ -195,7 +195,7 @@ VNUM ObjectManager::GetFreeRoomVnum(VNUM min, VNUM max, VNUM num)
     int i = min;
     if (num)
         {
-            return (RoomExists(num) ? num : 0);
+            return (!RoomExists(num) ? num : 0);
```

Comment:
Unless my understanding of this system is incorrect, I assume `getFreeRoomVnum` checks to see if a room number was specified (eg. its not the default 0 value). If so, check to see if the room doesn't exist. if it doesn't, return that number back to the calling function as valid (you can assign it to a room).